modules/cdc-ext/src/main/java/org/apache/ignite/cdc/postgresql/IgniteToPostgreSqlCdcConsumer.java

```diff
@@ -0,0 +1,251 @@
+
+        applier = new IgniteToPostgreSqlCdcApplier(dataSrc, batchSize, log);
+
+        MetricRegistryImpl mreg = (MetricRegistryImpl)reg;
+
+        this.evtsCnt = mreg.longMetric(EVTS_SENT_CNT, EVTS_SENT_CNT_DESC);
+        this.lastEvtTs = mreg.longMetric(LAST_EVT_SENT_TIME, LAST_EVT_SENT_TIME_DESC);
+
+        if (log.isInfoEnabled())
+            log.info("CDC Ignite to PostgreSQL start-up [cacheIds=" + cachesIds + ']');
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean onEvents(Iterator<CdcEvent> events) {
+        Iterator<CdcEvent> filtered = F.iterator(
+            events,
+            F.identity(),
+            true,
+            evt -> !onlyPrimary || evt.primary(),
+            evt -> cachesIds.contains(evt.cacheId()));
+
+        long evtsSent = applier.applyEvents(filtered);
+
+        if (evtsSent > 0) {
+            evtsCnt.add(evtsSent);
+            lastEvtTs.value(System.currentTimeMillis());
+
+            if (log.isInfoEnabled())
+                log.info("Events applied [evtsApplied=" + evtsCnt.value() + ']');
+        }
+
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onTypes(Iterator<BinaryType> types) {
+        types.forEachRemaining(e -> {
+            // Just skip. Handle of cache events not supported.
+        });
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onMappings(Iterator<TypeMapping> mappings) {
+        mappings.forEachRemaining(e -> {
+            // Just skip. Handle of cache events not supported.
+        });
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onCacheChange(Iterator<CdcCacheEvent> cacheEvents) {
+        Iterator<CdcCacheEvent> filtered = F.iterator(
+            cacheEvents,
+            F.identity(),
+            true,
+            evt -> cachesIds.contains(evt.cacheId()));
+
+        long tablesCreated = applier.applyCacheEvents(filtered, createTables);
+
+        if (tablesCreated > 0 && log.isInfoEnabled())
+            log.info("Cache changes applied [tablesCreatedCnt=" + tablesCreated + ']');
+    }
+
+    /** {@inheritDoc} */
+    @Override public void onCacheDestroy(Iterator<Integer> caches) {
+        caches.forEachRemaining(e -> {
+            // Just skip. Handle of cache events not supported.
+        });
+    }
+
+    /** {@inheritDoc} */
+    @Override public void stop() {
+
+    }
+
+    /**
+     * Sets the datasource configuration for connecting to the PostgreSQL database.
+     *
+     * @param dataSrc Configured data source.
+     * @return {@code this} for chaining.
+     */
+    public IgniteToPostgreSqlCdcConsumer setDataSource(DataSource dataSrc) {
+        this.dataSrc = dataSrc;
+
+        return this;
+    }
+
+    /**
+     * Sets cache names to replicate.
+     *
+     * @param caches Cache names.
+     * @return {@code this} for chaining.
+     */
+    public IgniteToPostgreSqlCdcConsumer setCaches(Set<String> caches) {
+        this.caches = caches;
```

Comment:
Let's use `new HashSet()` here. The `caches` collections is provided by user, then we can't guarantee that it won't be changed in any moment.